packages/mui-material-next/migration.md

```diff
@@ -6,6 +6,58 @@ This is a reference guide on how to migrate from Material UI v5 to v6.
 
 This section lists all breaking changes related to components in v6 and how to address them.
 
+## Overarching changes
+
+These are the changes that apply to all components
+
+### Remove `components` and `componentsProps` props
+
+The deprecated `components` and `componentsProps` props are removed in v6.
+If you were using these, then you can use `slots` and `slotProps` props instead,
+which have the same functionality and API.
+Here's an example of the change using the Badge component:
+
+```diff
+ <Badge
+     badgeContent={4}
+-    components={{ badge: (props) => <span {...props}>slot test</span> }}
++    slots={{ badge: (props) => <span {...props}>slot test</span> }}
+-    componentsProps={{ badge: { id: "test-id" } }}
++    slotProps={{ badge: { id: "test-id" } }}
+ >
+   <MailIcon color="action" />
+ </Badge>
+```
+
+## Badge
+
+### Added and removed colors
+
+Color options were changed to match Material You:
+
+- `"default"`, `"info"`, `"success"`, and `"warning"` were removed and are no longer supported out-of-the-box
+- `"tertiary"` was added
+- `"error"` is now the default
+
+### Renamed variants
+
+Variants were renamed to match Material You:
+
+- `standard` renamed to `large`, which is still the default
+- `dot` renamed to `small`
```

Comment:
We use `size` as a name for the sizes. I propose keeping all variant values as deprecated to easy up the migration, and use a new `size` prop for this instead. We can deprecate the whole `variant` prop with an appropriate deprecation message.

Reply:
@mnajdova I agree with naming the prop `size`.

About keeping the `variant` prop, I think it would be better to remove it right away, because:

- Migrating can be covered with a codemod
- Keeping it would complicate and bloat the code 

What do you think?

Reply:
I feel like we can likely get more people to try it early if we don't introduce many breaking changes with the first version, but happy to experiment with what you suggest.

I see two ways going forward:

- deprecate props and move forward with minimal breaking changes
- introduce breaking changes together with a codemod for them at the moment they are implemented

So we should either, add a codemod now, or deprication, one of these two needs to be implemented.